ipam/allocator.go

```diff
@@ -122,19 +112,68 @@ func (a *Allocator) updateBitMasks(aSpace *addrSpace) error {
 func (a *Allocator) checkConsistency(as string) {
 	// Retrieve this address space's configuration and bitmasks from the datastore
 	a.refresh(as)
+	a.Lock()
 	aSpace, ok := a.addrSpaces[as]
+	a.Unlock()
 	if !ok {
 		return
 	}
 	a.updateBitMasks(aSpace)
+	aSpace.Lock()
```

Comment:
I need to iterate through the address space's map of pools. I don't think I can avoid this lock.
Anyway, the consistency check is run either at allocator creation or during datastore update. There won't not be other activity on the datastore during these two phases.